src/js/client/editprefset-productList-widget.js

```diff
@@ -0,0 +1,149 @@
+                    initial: "#common-filter-area"
+                },
+                model: {
+                    filterText: "{productListWidget}.model.productListFilter"
+                }
+            }
+        }
+    },
+    lunrIndex: null,
+    invokers: {
+        renderInitialMarkup: {
+            func: "{that}.renderMarkup",
+            args: ["initial", "{that}.options.templates.initial", "{that}.model"]
+        },
+        updateLunrIndex: {
+            funcName: "gpii.devpmt.productList.updateProductListLunrIndex",
+            args: ["{that}"]
+        },
+        filterProductList: {
+            funcName: "gpii.devpmt.productList.filterProductList",
+            args: ["{that}", "{that}.dom.productListItems", "{that}.model.productListFilter"]
+        },
+        // Used when clicking on the product, or otherwise activating it from the list.
+        selectProduct: {
+            funcName: "gpii.devpmt.productList.selectProduct",
+            args: ["{that}", "{gpii.devpmt.editPrefs}", "{arguments}.0"]
+        }
+    },
+    events: {
+        activateProduct: null
+    },
+    listeners: {
+        "onMarkupRendered": [
+            {
+                "this": "{that}.dom.productListLinks",
+                "method": "click",
+                args: ["{that}.selectProduct"]
+            }
+        ]
+    }
+});
+
+gpii.devpmt.productList.selectProduct = function (that, devpmt, event) {
+    var appid = event.currentTarget.dataset.appid;
+    that.events.activateProduct.fire(appid);
+    // TODO Fix up the create order or figure out how to add a delayed listener,
+    // such that devpmt can listen for this, since it can't create a listener, since
+    // this component is created after it is finished being created.
+    devpmt.openAddProductDialog(appid);
+};
+
+/**
+ * Filters the products in the list, usually runs on a model listener for when
+ * a user updates the filter text.
+ *
+ * @param that
+ * @param productListItems
+ * @param productListFilter
+ */
+gpii.devpmt.productList.filterProductList = function (that, productListItems, productListFilter) {
+    if (!that.lunrIndex) {
+        return;
+    }
+    productListItems.hide();
+
+    var results = gpii.devpmt.lunrListFilterSearch(that.lunrIndex, productListFilter);
+
+    // TODO Remove this double loop
+    fluid.each(productListItems, function (product) {
+        fluid.each(results, function (result) {
+            if (result.ref === product.dataset.appid) {
+                $(product).show();
+            }
+        });
+    });
+};
+
+/**
+ * Typically runs in response to a change in the list of products in the
+ * list so that the lunr index can either be rebuilt or updated.
+ *
+ * @param that
+ */
+gpii.devpmt.productList.updateProductListLunrIndex = function (that) {
+    var lunrIndex = lunr(function () {
+        var idx = this;
+        this.ref("id");
+        this.field("name");
+        this.field("appId");
+        this.b(0.01);
```

Comment:
What is this undocumented member named b?

Reply:
it's the "field length normalisation" field in lunr.js which has been documented here now